adapter/internal/discovery/xds/marshaller.go

```diff
@@ -659,14 +659,22 @@ func marshalApplicationPolicy(policy *types.ApplicationPolicy) *subscription.App
 }
 
 func marshalSubscriptionPolicy(policy *types.SubscriptionPolicy) *subscription.SubscriptionPolicy {
+	if policy.DefaultLimit == nil || policy.DefaultLimit.RequestCount == nil {
+		policy.DefaultLimit = &types.SubscriptionDefaultLimit{
+			RequestCount: &types.SubscriptionRequestCount{
+				RequestCount: policy.RateLimitCount,
+				TimeUnit:     policy.RateLimitTimeUnit,
+			},
+		}
```

Comment:
We discussed this earlier too. :) Also in a recent call. This is for the test cases.

Reply:
Instead, shall we assign -1 and min (unlimited policy configs) and let's add a comment mentioning, that we are doing this to avoid integration test failure because we are using onprem APIM?